src/mantle/framework/class-bootloader.php

```diff
@@ -68,13 +76,157 @@ public static function set_instance( ?Bootloader $instance = null ): void {
 		static::$instance = $instance;
 	}
 
+	/**
+	 * Clear the instance of the manager.
+	 */
+	public static function clear_instance(): void {
+		static::$instance = null;
+	}
+
 	/**
 	 * Constructor.
 	 *
 	 * @param Contracts\Application|null $app Application instance.
+	 * @param string|null                $base_path Base path for the application.
 	 */
-	public function __construct( protected ?Contracts\Application $app = null ) {
+	public function __construct( protected ?Contracts\Application $app = null, ?string $base_path = null ) {
 		static::set_instance( $this );
+
+		$this
+			->with_application( $app ?? new Application( $base_path ) )
+			->with_kernels()
+			->with_exception_handler();
+	}
+
+	/**
+	 * Set the application instance to be booted.
+	 *
+	 * @param Contracts\Application $app Application instance.
+	 */
+	public function with_application( Contracts\Application $app ): static {
+		$this->app = $app;
+
+		return $this;
+	}
+
+	/**
+	 * Merge additional configuration to the existing configuration.
+	 *
+	 * Configuration passed will be merged recursively with the existing
+	 * configuration after all application configuration has been loaded.
+	 *
+	 * @param array<string, array<mixed>> $config Configuration to merge.
+	 */
+	public function with_config( array $config ): static {
+		Load_Configuration::merge( $config );
+
+		return $this;
+	}
+
+	/**
+	 * Bind the application with the default kernels.
+	 *
+	 * @throws \InvalidArgumentException If the console or HTTP kernel does not implement the correct interface.
+	 *
+	 * @param class-string<Contracts\Console\Kernel>|null $console Console kernel class.
+	 * @param class-string<Contracts\Http\Kernel>|null    $http    HTTP kernel class.
+	 */
+	public function with_kernels( string $console = null, string $http = null ): static {
+		if ( $console && ! in_array( Contracts\Console\Kernel::class, class_implements( $console ), true ) ) {
```

Comment:
Would a `! class_exists( $handler )` also make sense here? Or would this already be caught? Same question for `with_exception_handler` method below.